### PR TITLE
Show alignment in reprs and document alignment

### PR DESCRIPTION
**Pull request by jl-wynen**
This is the last step for implementing the alignment flag as per #3109.

### CODE REVIEW

File: src/scipp/compat/dict.py [hunk 1]
```diff
@@ -131,14 +133,15 @@ def from_dict(dict_obj: dict) -> VariableLike:
...
-        {"dims", "values", "variances", "unit", "dtype", "shape"}
+        {"dims", "values", "variances", "unit", "dtype", "shape", "aligned"}
     ):
         # Case of a Variable-like dict (most-likely)
         return _dict_to_variable(dict_obj)
     else:
         # Case of a Dataset-like dict
         out = Dataset()
         for key, item in dict_obj.items():
+            print(key, item)
```
Comment (SimonHeybrock) [inline src/scipp/compat/dict.py:144]:
```suggestion
```

File: lib/python/variable.cpp [hunk 2]
```diff
@@ -108,8 +98,44 @@ For *binned* coordinates of a binned data array ``da``,
...
+      .def(
+          "set_aligned",
+          [](Variable &self, const bool aligned) { self.set_aligned(aligned); },
+          R"(Set the alignment flag of the variable.
+
+Attention
+---------
+This is likely not the function you want to use as it may only affect
+a temporary variable and not the actual coordinate of a data array.
+Use ``da.coords.set_aligned(name, alignment)`` instead.
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply lib/python/variable.cpp:112]: To simplify hdf5 I/O and dict conversion. Without it, those functions could not be implemented in the current modular way. I.e. the handlers for coords would have to either infect the flag into the constituent variable groups/subdicts or emit an additional mapping as a dataset/dict.
Minor: Saving and loading a variable on its own currently preserves alignment even outside of coords.

[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply lib/python/variable.cpp:112]: I do worry a bit about making this part of the public interface. The docstring will not really prevent misuse and hours of searching for problems.

Can it be solved by making the `aligned` flag a constructor argument of `Variable`?

[the remark this answers is not in the document]
Comment (jl-wynen) [reply lib/python/variable.cpp:112]: Should be possible. So would this go into `Variable.__init__`, `array`, and `scalar`? Or should we not even provide it in the dedicated functions?

[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply lib/python/variable.cpp:112]: Maybe just `Variable`, but not the functions?

File: lib/python/variable_init.cpp [hunk 3]
```diff
@@ -329,5 +334,7 @@ Parameters
    Type of the variable's elements. Is deduced from other arguments
    in most cases. Defaults to ``sc.DType.float64`` if no deduction is
    possible.
+aligned:
```
Comment (SimonHeybrock) [inline lib/python/variable_init.cpp:337]: Unrelated to this change, but I just realize that sphinx-autodoc-typehints will not help us here (or any other function without Python wrapper)? It looks like the types are not documented: https://scipp.github.io/generated/classes/scipp.Variable.html#scipp.Variable.__init__

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**